docs/index.rst

```diff
@@ -666,6 +681,14 @@ You can customize the instantiator in several ways:
         })
     ;
 
+.. warning::
+
+    The ``instantiateWith()`` method fully replaces the default instantiation
+    and object hydration system. Attributes defined in the ``defaults()`` method,
+    as well as any states defined with the ``with()`` method, **will not be
+    applied automatically**. However, they are available as arguments to the
+    ``instantiateWith()`` callable.
```

Comment:
sorry but I still don't agree :sweat_smile:  

what do you mean by "will not be applied automatically"? I feel this is confusing. 

If you call something like: `$factory->instantiateWith(Instantiator::withoutConstructor())`, the attributes are "applied automatically". They are ineeded passed to the callable, which in that case is the class `Instantiator`, but this all happen under the hood.

What was your initial problem that led you to make this change in the doc?

Reply:
> what do you mean by "will not be applied automatically"? I feel this is confusing.

When using a totally custom callback, the hydrator is not used. I think Javier expected it to just instantiate the object but still run in through the hydrator.

At one point I did have a PR to split these two concepts.

Reply:
ok, I understand better now!

you tried the version `instantiateWith(callable(...))`, which is the one I just never use. This one should only be used in rare cases when Foundry is not able to instantiate/hydrate the object. Nonetheless, it is not rare to use  `instantiateWith()` with an `Instantiator` object as an argument.

I think the solution for the docs, would be to separate the last special case from the code block above your comment, and add the warning only for this case

> In defaults you can pass both constructor arguments and properties!!

if this is not crystal clear in the docs, this may be highlighted!